pkg/apis/pipeline/v1beta1/param_types.go

```diff
@@ -284,12 +290,48 @@ func validateParameterInOneOfMatrixOrParams(matrix []Param, params []Param) (err
 	return errs
 }
 
-func validateStringVariable(value, prefix string, stringVars sets.String, arrayVars sets.String) *apis.FieldError {
+// validateStringVariableP validates param variable usage in string value
+// This will allow referencing the whole object/array value. example:
+// params:
+// 	- name: arg
+// 		value: $(params.myObject[*])
+// see https://github.com/tektoncd/pipeline/issues/4879#issuecomment-1133120522 for background
```

Comment:
I think there's not so much context in this issue comment that it makes sense to link to it; it can just be explained here. I would also focus more on what this function does based on its inputs, and if you're going to include example syntax make sure it's formatted correctly.